typewritten.zsh

```diff
@@ -150,7 +150,9 @@ tw_setup() {
 
   zmodload zsh/zle
   autoload -Uz add-zsh-hook
-  add-zsh-hook precmd tw_fix_cursor
+  if [ "$TYPEWRITTEN_CURSOR" != "terminal default" ]; then
```

Comment:
I think just `"terminal"` would be the more accurate term, semantically. If I change my cursor in my terminal emulator, then my terminal cursor is no longer `default`.

Reply:
of course, no problem, `terminal` is more accurate

Reply:
Cool, update it and we can merge. The documentation also needs to be updated for the `TYPEWRITTEN_CURSOR` section. You can update it, or leave it to me and I'll update your PR np.